torch/csrc/distributed/rpc/init.cpp

```diff
@@ -409,31 +410,40 @@ If the future completes with an error, an exception is thrown.
          const py::object& userCallable,
          const py::tuple& argsTuple,
          const py::dict& kwargsDict) {
-        DCHECK(PyGILState_Check());
+        DCHECK(!PyGILState_Check());
         // No need to catch exception here, if function can not be found,
         // exception will be thrown in get_function() call; if args do not match
         // with function schema, exception will be thrown in
         // createStackForSchema() call.
-        auto qualifiedNameStr = c10::QualifiedName(
-            py::cast<std::string>(py::module::import("torch.jit")
-                                      .attr("_qualified_name")(userCallable)));
-        auto qualifiedName = c10::QualifiedName(qualifiedNameStr);
-        auto functionSchema = PythonRpcHandler::getInstance()
-                                  .jitCompilationUnit()
-                                  ->get_function(qualifiedName)
-                                  .getSchema();
-        Stack stack = torch::jit::createStackForSchema(
-            functionSchema,
-            argsTuple.cast<py::args>(),
-            kwargsDict.cast<py::kwargs>(),
-            c10::nullopt);
-        py::gil_scoped_release release;
+        {
+          py::gil_scoped_acquire acquire;
+          c10::QualifiedName qualifiedName =
+              c10::QualifiedName(py::cast<std::string>(
+                  py::module::import("torch.jit")
+                      .attr("_qualified_name")(userCallable)));
+        }
```

Comment:
we can move this import to python rpc handler as well to avoid import every request